components/server/src/github/github-repository-provider.ts

```diff
@@ -230,6 +230,77 @@ export class GithubRepositoryProvider implements RepositoryProvider {
         return repos;
     }
 
+    public async searchRepos(user: User, searchString: string): Promise<RepositoryInfo[]> {
+        const logCtx = { userId: user.id, searchString };
+
+        // TODO: look to try graphql api to reduce response payload size
+        // search personal repos
+        const userSearch = this.github.run(user, async (api) => {
+            return api.search.repos({
+                q: `"${searchString}" in:name user:@me sort:updated`,
+                sort: "updated",
+                per_page: 10,
+            });
+        });
+
+        // Attach an error handler to log error and not throw
+        userSearch.catch((err) => {
+            log.warn(logCtx, "Error searching user repos", err);
+        });
+
+        // TODO: Look at using graphql query to grab all orgs w/ some kind of latest activity date
+        // We could then sort in memory and grab the 5 most recent orgs
+        // Find all orgs user belongs to
+        const orgs = await this.github
+            .run(user, async (api) => {
+                return api.orgs.listMembershipsForAuthenticatedUser({
+                    state: "active",
+                    per_page: 5,
```

Comment:
I tried to find a repo that I have access to but couldn't find it. Probably because it is not among those 5 orgs. Would it be to slow to search all orgs?

Reply:
Since we aim to already provide the "active" repositories with the other query that doesn't pass in a search, I believe this one should rather be complete in what I can find and could compromise a bit (if not too much) on performance.

Reply:
Yah I agree. I'll play around with it a bit more.